src/skydb_internal.ts

```diff
@@ -0,0 +1,194 @@
+import {
+  deriveEncryptedFileKeyEntropy,
+  deriveEncryptedFileTweak,
+  decryptJSONFile,
+  ENCRYPTED_JSON_RESPONSE_VERSION,
+  EncryptedJSONResponse,
+  getOrCreateSkyDBRegistryEntry,
+  JsonData,
+  SkynetClient,
+  encryptJSONFile,
+  MAX_REVISION,
+  RegistryEntry,
+  signEntry,
+} from "skynet-js";
+
+import { deriveRootPathSeed, genKeyPairFromSeed } from "./crypto";
+import { deriveEncryptedPathSeedForRoot } from "./encrypted_files";
+import { log, validateObject, validateString } from "./util";
+
+/**
+ * Gets Encrypted JSON at the given path through MySky.
+ *
+ * @param client - The Skynet client.
+ * @param seed - The root MySky user seed.
+ * @param path - The data path.
+ * @returns - An object containing the decrypted json data.
+ * @throws - Will throw if the user does not have Hidden Read permission on the path.
+ */
+export async function getJSONEncryptedInternal(
+  client: SkynetClient,
+  seed: Uint8Array,
+  path: string
+): Promise<EncryptedJSONResponse> {
+  log("Entered getJSONEncryptedInternal");
+
+  validateString("path", path, "parameter");
+
+  const { publicKey } = genKeyPairFromSeed(seed);
+  const pathSeed = await getEncryptedPathSeedInternal(seed, path, false);
+
+  // Fetch the raw encrypted JSON data.
+  const dataKey = deriveEncryptedFileTweak(pathSeed);
+  const opts = { hashedDataKeyHex: true };
+  log("Calling getRawBytes");
+  const { data } = await client.db.getRawBytes(publicKey, dataKey, opts);
+  if (data === null) {
+    return { data: null };
+  }
+
+  const encryptionKey = deriveEncryptedFileKeyEntropy(pathSeed);
+  const json = decryptJSONFile(data, encryptionKey);
+
+  return { data: json };
+}
+
+/**
+ * Sets Encrypted JSON at the given path through MySky.
+ *
+ * @param client - The Skynet client.
+ * @param seed - The root MySky user seed.
+ * @param path - The data path.
+ * @param json - The json to encrypt and set.
+ * @returns - An object containing the original json data.
+ * @throws - Will throw if the user does not have Hidden Write permission on the path.
+ */
+export async function setJSONEncryptedInternal(
+  client: SkynetClient,
+  seed: Uint8Array,
+  path: string,
+  json: JsonData
+): Promise<EncryptedJSONResponse> {
+  log("Entered setJSONEncryptedInternal");
+
+  validateString("path", path, "parameter");
+  validateObject("json", json, "parameter");
+
+  const { publicKey } = genKeyPairFromSeed(seed);
+  const pathSeed = await getEncryptedPathSeedInternal(seed, path, false);
+  const dataKey = deriveEncryptedFileTweak(pathSeed);
+  const opts = { hashedDataKeyHex: true };
+
+  // Immediately fail if the mutex is not available.
+  return await client.db.revisionNumberCache.withCachedEntryLock(
```

Comment:
do we care about these linting errors here? `Property 'revisionNumberCache' does not exist on type` looks li ke the database isn't properly typed here